### PR TITLE
HTML CI

### PR DESCRIPTION
**Pull request by Schwad**
* Introduces HTML fixtures for webview. They are generated off of our *top-level* Ruby example files. Generate or regenerate with `bundle exec rake test:regen_fixtures`
* Check that your changes have not altered the html with `bundle exec rake test:check_html_fixtures`
  * Your change may well result in new HTML and that is okay. But this will make it easier to review the actual HTML diff
* Introduces CI check after all prior tests to ensure that developers make this check.
  * Example of a failure output on a CI run here: https://github.com/scarpe-team/scarpe/actions/runs/7159731189/job/19493310868

### CODE REVIEW

File: lacci/test/test_oval.rb [hunk 1]
```diff
@@ -26,6 +26,7 @@ def test_oval_with_height
       end
     SHOES_APP
       ov = oval()
+      binding.irb
```
Comment (noahgibbs) [inline lacci/test/test_oval.rb:29]: Was this on purpose?

Comment (Schwad) [reply lacci/test/test_oval.rb:29]: No. No it is not. Thanks noah, this was a bigger pr

Comment (Schwad) [reply lacci/test/test_oval.rb:29]: and I may have pushed more quickly as I'm cooking

File: test/wv/html_fixtures/animate.html [hunk 2]
```diff
@@ -0,0 +1,12 @@
+<div id="2" style="display:flex;flex-direction:row;flex-wrap:wrap;align-content:flex-start;justify-content:flex-start;align-items:flex-start;width:100%;height:100%">
```
Comment (noahgibbs) [inline test/wv/html_fixtures/animate.html:1]: Pretty-printing the HTML is a nice touch. This looks good.

File: tasks/check_html_fixtures.rb [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+  end
+
+  #Handle the last remaining hunk
+  output << oldhunk.diff(:unified, "#{file_name}:") << "\n"
+end
+
+file_names.each do |file_name|
+  output = ""
+  pid = nil
+  command = "bundle exec ./exe/scarpe examples/#{file_name} --dev --debug"
```
Comment (noahgibbs) [inline tasks/check_html_fixtures.rb:69]: Ah, you're just running with --debug and reading the HTML off the console. Devious!

File: tasks/check_html_fixtures.rb [hunk 4]
```diff
@@ -0,0 +1,125 @@
+# frozen_string_literal: false
+require 'diff/lcs'
+require 'diff/lcs/hunk'
+require 'fileutils'
+require "htmlbeautifier"
+
+
+# Get top-level Ruby examples
+file_paths = Dir["examples/*.rb"]
```
Comment (noahgibbs) [inline tasks/check_html_fixtures.rb:9]: I think you're lucky you picked this exact set of examples. Anything with a local asset server is going to cause you trouble -- its port number is going to change run-to-run. Worse yet, we *want* its port number to change run-to-run. Though we can probably make it work somehow.

**Comment by noahgibbs** [comment]
Getting a couple of failures on button rendering not matching up with property order... Not sure why. Also an error on local_font.rb because the CI full file path for the font doesn't match the one from @schwad's machine. There are a few possible ways to fix that...

File: tasks/check_html_fixtures.rb (hunk 3 above)
Comment (Schwad) [reply tasks/check_html_fixtures.rb:69]: I built this PR one step at a time starting with that. debug one example. multiple examples. save output. make it a task. compare. make output pretty. put in ci. , iterate, iterate, iterate, to what we have.

**Comment by Schwad** [comment]
> Getting a couple of failures on button rendering not matching up with property order... Not sure why. Also an error on local_font.rb because the CI full file path for the font doesn't match the one from @Schwad's machine. There are a few possible ways to fix that...

I'll dig into this and see the best way to resolve

File: tasks/check_html_fixtures.rb (hunk 4 above)
Comment (Schwad) [reply tasks/check_html_fixtures.rb:9]: Yeah, good to know. Happy to fix forward ways to make it increasingly resilient. Though it's an extra headwind being a new ci step I still want to keep it friendly as possible and there are multiple things we can look at too going forward :)

**Comment by Schwad** [comment]
Another cool thing, like, is being able to drop this raw HTML into your browser and just "look" at it. it's nothing extravagant and maybe I need another higher level div in there but I really like it

**Comment by Schwad** [comment]
I've added support for

`# html_ci: false` 

for examples that have issues until we resolve them.

**Review by noahgibbs — APPROVED** [review]

**Pull request merged**